homeassistant/components/fritzbox/config_flow.py

```diff
@@ -36,20 +41,21 @@
 RESULT_SUCCESS = "success"
 
 
-class FritzboxConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+class FritzboxConfigFlow(ConfigFlow, domain=DOMAIN):
     """Handle a AVM FRITZ!SmartHome config flow."""
 
     VERSION = 1
 
-    def __init__(self):
+    def __init__(self) -> None:
         """Initialize flow."""
-        self._entry = None
-        self._host = None
-        self._name = None
-        self._password = None
-        self._username = None
-
-    def _get_entry(self):
+        self._entry: ConfigEntry | None = None
+        self._host: str | None = None
+        self._name: str | None = None
+        self._password: str | None = None
+        self._username: str | None = None
+
+    def _get_entry(self) -> FlowResult:
+        assert self._name is not None
```

Comment:
Pass it as argument. Same in `_update_entry`.
```suggestion
    def _get_entry(self, name: str) -> FlowResult:
```